rest_api/controller/file_upload.py

```diff
@@ -62,27 +63,27 @@ def upload_file_to_document_store(
 
         if file.filename.split(".")[-1].lower() == "pdf":
             pdf_converter = PDFToTextConverter(
-                remove_numeric_tables=remove_numeric_tables,
-                remove_whitespace=remove_whitespace,
-                remove_empty_lines=remove_empty_lines,
-                remove_header_footer=remove_header_footer,
-                valid_languages=valid_languages,
+                remove_numeric_tables=remove_numeric_tables, valid_languages=valid_languages
             )
             document = pdf_converter.convert(file_path)
         elif file.filename.split(".")[-1].lower() == "txt":
             txt_converter = TextConverter(
-                remove_numeric_tables=remove_numeric_tables,
-                remove_whitespace=remove_whitespace,
-                remove_empty_lines=remove_empty_lines,
-                remove_header_footer=remove_header_footer,
-                valid_languages=valid_languages,
+                remove_numeric_tables=remove_numeric_tables, valid_languages=valid_languages,
             )
             document = txt_converter.convert(file_path)
         else:
             raise HTTPException(status_code=415, detail=f"Only .pdf and .txt file formats are supported.")
 
-        document_to_write = {TEXT_FIELD_NAME: document["text"], "name": file.filename}
-        document_store.write_documents([document_to_write])
+        document = {TEXT_FIELD_NAME: document["text"], "name": file.filename}
+
+        preprocessor = PreProcessor(
+            clean_whitespace=remove_whitespace,
+            clean_header_footer=remove_header_footer,
+            clean_empty_lines=remove_empty_lines,
+        )
```

Comment:
How about adding the split params (`split_by` etc)here, too? They became quite important with DPR etc.